main.go

```diff
@@ -200,23 +193,33 @@ func main() {
 		log.Printf("Dumped system information to: %s", basePath+".tar.gz")
 	}()
 
+	log.Print("Starting RHMAP System Dump Tool...")
 	log.Println("Running tasks...")
 
-	// Run at most N tasks in parallel, and wait for all of them to
-	// complete.
+	tasks := GetAllTasks(basePath)
+	e := make(chan error)
+
 	var wg sync.WaitGroup
-	sem := make(chan struct{}, *maxParallelTasks)
-	for _, task := range tasks {
-		task := task
-		wg.Add(1)
-		sem <- struct{}{}
+	wg.Add(*parallelTasks)
+	for i := 0; i < *parallelTasks; i++ {
 		go func() {
-			defer wg.Done()
-			task()
```

Comment:
@philbrookes notice here that we used to completely ignore task errors!
You can see below that now we log errors to the screen and to the "dump.log" that is part of the archive we produce.